crypto/fipsmodule/ec/ec_test.cc

```diff
@@ -182,6 +182,259 @@ TEST(ECTest, Encoding) {
+        NID_secp384r1
+    },
+    {
+        kP521PublicKey_compressed, sizeof(kP521PublicKey_compressed),
+        POINT_CONVERSION_COMPRESSED,
+        kP521PublicKey_compressed, sizeof(kP521PublicKey_compressed),
+        NID_secp521r1
+    },
+    /* Test 3: decode compressed |EC_KEY|, and then encode with uncompressed |conv_form|. */
+    {
+        kP224PublicKey_compressed, sizeof(kP224PublicKey_compressed),
+        POINT_CONVERSION_UNCOMPRESSED,
+        kP224PublicKey_uncompressed, sizeof(kP224PublicKey_uncompressed),
+        NID_secp224r1
+    },
+    {
+        kP256PublicKey_compressed, sizeof(kP256PublicKey_compressed),
+        POINT_CONVERSION_UNCOMPRESSED,
+        kP256PublicKey_uncompressed, sizeof(kP256PublicKey_uncompressed),
+        NID_X9_62_prime256v1
+    },
+    {
+        kP384PublicKey_compressed, sizeof(kP384PublicKey_compressed),
+        POINT_CONVERSION_UNCOMPRESSED,
+        kP384PublicKey_uncompressed, sizeof(kP384PublicKey_uncompressed),
+        NID_secp384r1
+    },
+    {
+        kP521PublicKey_compressed, sizeof(kP521PublicKey_compressed),
+        POINT_CONVERSION_UNCOMPRESSED,
+        kP521PublicKey_uncompressed, sizeof(kP521PublicKey_uncompressed),
+        NID_secp521r1
+    },
+    /* Test 4: decode uncompressed |EC_KEY|, and then encode with compressed |conv_form|. */
+    {
+        kP224PublicKey_uncompressed, sizeof(kP224PublicKey_uncompressed),
+        POINT_CONVERSION_COMPRESSED,
+        kP224PublicKey_compressed, sizeof(kP224PublicKey_compressed),
+        NID_secp224r1
+    },
+    {
+        kP256PublicKey_uncompressed, sizeof(kP256PublicKey_uncompressed),
+        POINT_CONVERSION_COMPRESSED,
+        kP256PublicKey_compressed, sizeof(kP256PublicKey_compressed),
+        NID_X9_62_prime256v1
+    },
+    {
+        kP384PublicKey_uncompressed, sizeof(kP384PublicKey_uncompressed),
+        POINT_CONVERSION_COMPRESSED,
+        kP384PublicKey_compressed, sizeof(kP384PublicKey_compressed),
+        NID_secp384r1
+    },
+    {
+        kP521PublicKey_uncompressed, sizeof(kP521PublicKey_uncompressed),
+        POINT_CONVERSION_COMPRESSED,
+        kP521PublicKey_compressed, sizeof(kP521PublicKey_compressed),
+        NID_secp521r1
+    }
+};
+
+class ECPublicKeyTest : public testing::TestWithParam<ECPublicKeyTestInput> {};
+
+// This is to test |EC_KEY| decode using |o2i_ECPublicKey| and encode using
+// |i2o_ECPublicKey|.
+TEST_P(ECPublicKeyTest, DecodeAndEncode) {
+  const auto &param = GetParam();
+  const auto input_key = param.input_key;
+  const auto input_key_len = param.input_key_len;
+  const auto encode_conv_form = param.encode_conv_form;
+  const auto expected_output_key = param.expected_output_key;
+  const auto expected_output_key_len = param.expected_output_key_len;
+  const auto nid = param.nid;
+  // Generate |ec_key|.
+  EC_KEY *ec_key = EC_KEY_new();
+  ASSERT_TRUE(ec_key);
+  bssl::UniquePtr<EC_KEY> ec_key_ptr(ec_key);
```

Comment:
Is `ec_key_ptr` not unused? Or this will automatically free the `EC_KEY` object?

Reply:
> Or this will automatically free the EC_KEY object?

Yes, `bssl::UniquePtr` is used to own the lifecycle of `ec_key`. After this test method scope, the pointer will be released.